gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionExplorer.java

```diff
@@ -34,83 +32,62 @@
 import java.util.concurrent.TimeUnit;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
-import org.immutables.value.Value;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
-import org.jsoup.nodes.Element;
-import org.jsoup.select.Elements;
+import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class RepositoryExplorer {
-    private static final Logger log = LoggerFactory.getLogger(RepositoryExplorer.class);
+public class VersionExplorer {
+    private static final Logger log = LoggerFactory.getLogger(VersionExplorer.class);
 
     private static final Pattern UNSTABLE_VERSION_PATTERN = Pattern.compile(
             ".*(-rc(-?\\d+)?|-SNAPSHOT|-M\\d+|-alpha(-?\\d+)?|-beta(-?\\d+)?)$", Pattern.CASE_INSENSITIVE);
 
-    private final Cache<CacheKey, Set<GroupPartOrPackageName>> folderCache = Caffeine.newBuilder()
-            .expireAfterWrite(10, TimeUnit.MINUTES)
-            .maximumSize(100)
-            .build();
+    // In general, we don't want to be caching version data as it changes often. However, for wildcard complete it
+    // can be very expensive to repeatedly get data that realistically doesn't change on a second by second basis so
+    // having a short-lived cache is okay
+    private final AsyncLoadingCache<String, Set<DependencyVersion>> shortLivedVersionCache = Caffeine.newBuilder()
+            .expireAfterWrite(2, TimeUnit.MINUTES)
+            .maximumSize(10000)
+            .buildAsync(this::fetchAndParseFromUrl);
 
-    public final Set<GroupPartOrPackageName> getGroupPartOrPackageName(DependencyGroup group, String url) {
-        CacheKey cacheKey = CacheKey.of(url, group);
-        Set<GroupPartOrPackageName> folders = folderCache.get(cacheKey, key -> {
-            Set<GroupPartOrPackageName> loadedFolders = loadFolders(key.group(), url);
-            return loadedFolders.isEmpty() ? null : loadedFolders;
-        });
+    public record GroupAndDep(DependencyGroup group, DependencyName dependencyPackage, String url) {}
```

Comment:
This was just a random name I gave this when hacking it together. I'd probably call it something better - PackageInRepo or something like that?

`url` should be something like `repositoryUrl` - perhaps even with it's own type?

Reply:
renamed and created a repo url type